packages/eslint-plugin/src/rules/prefer-return-this-type.ts

```diff
@@ -148,24 +148,27 @@ export default createRule({
       }
     }
 
+    function checkProperty(
+      node: TSESTree.AccessorProperty | TSESTree.PropertyDefinition,
+    ): void {
+      if (
+        !(
+          node.value?.type === AST_NODE_TYPES.FunctionExpression ||
+          node.value?.type === AST_NODE_TYPES.ArrowFunctionExpression
+        )
+      ) {
+        return;
+      }
+
+      checkFunction(node.value, node.parent.parent);
+    }
+
     return {
+      'ClassBody > AccessorProperty': checkProperty,
       'ClassBody > MethodDefinition'(node: TSESTree.MethodDefinition): void {
-        checkFunction(node.value, node.parent.parent as ClassLikeDeclaration);
-      },
-      'ClassBody > PropertyDefinition'(
-        node: TSESTree.PropertyDefinition,
-      ): void {
-        if (
-          !(
-            node.value?.type === AST_NODE_TYPES.FunctionExpression ||
-            node.value?.type === AST_NODE_TYPES.ArrowFunctionExpression
-          )
-        ) {
-          return;
-        }
-
-        checkFunction(node.value, node.parent.parent as ClassLikeDeclaration);
+        checkFunction(node.value, node.parent.parent);
```

Comment:
The type assertion seems unnecessary since https://github.com/typescript-eslint/typescript-eslint/pull/9560.

Reply:
I wonder if this can be caught by `no-unnecessary-type-assertion` 🤔. Since `ClassLikeDeclaration` is a different type alias, the comparison seems to fail (even though `ClassLikeDeclaration` is assigned the same type as the un-asserted type).

Reply:
Yeah that would be nice. My hunch is it's some nuance about generic functions having type parameters inferred from types of arguments.